packages/stark-ui/src/modules/date-range-picker/components/date-range-picker.component.ts

```diff
@@ -24,91 +38,151 @@ const componentName = "stark-date-range-picker";
+	private _startBeforeEndVal: ValidatorFn = ({ value }) => {
+		return value instanceof Date && this.endDate instanceof Date && value.getTime() > this.endDate.getTime()
+			? { startBeforeEnd: true }
+			: null;
+	};
 
 	/**
-	 * Minimum date of the end date picker
+	 * Source Date to be bound to the start datepicker model
 	 */
 	@Input()
-	public endMinDate?: Date;
+	public get startDate(): Date | undefined {
+		return this._startDate.value || undefined;
+	}
+
+	public set startDate(value: Date | undefined) {
+		this._startDate.setValue(value);
+	}
+
+	public get startDateFormControl(): FormControl {
+		return this._startDate;
+	}
+
+	private _startDate = new FormControl(null, this._startBeforeEndVal);
 
 	/**
-	 * Whether the datepickers are disabled
+	 * Label to be displayed in the end datepicker
 	 */
 	@Input()
-	public isDisabled?: boolean;
+	public startDateLabel = "STARK.DATE_RANGE_PICKER.FROM";
 
 	/**
-	 * HTML "name" attribute of the element.
+	 * Minimum date of the start date picker
 	 */
 	@Input()
-	public rangePickerId = "";
+	public startMinDate?: Date;
 
 	/**
-	 * HTML "name" attribute of the element.
+	 * Maximum date of the start date picker
 	 */
 	@Input()
-	public rangePickerName = "";
+	public startMaxDate?: Date;
+
+	/*--- END-DATE CONFIGURATIONS ---*/
+
+	private _endAfterStartVal: ValidatorFn = ({ value }) => {
+		return value instanceof Date && this.startDate instanceof Date && value.getTime() < this.startDate.getTime()
+			? { endAfterStart: true }
+			: null;
+	};
 
 	/**
-	 * Source Date to be bound to the start datepicker model
+	 * Source Date to be bound to the end datepicker model
 	 */
+	@Input()
+	public get endDate(): Date | undefined {
+		return this._endDate.value || undefined;
+	}
+
+	public set endDate(value: Date | undefined) {
+		this._endDate.setValue(value || null);
+	}
+
+	public get endDateFormControl(): FormControl {
+		return this._endDate;
+	}
+
+	private _endDate = new FormControl(null, this._endAfterStartVal);
 
+	/**
+	 * Label to be displayed in the end datepicker
+	 */
 	@Input()
-	public startDate?: Date;
+	public endDateLabel = "STARK.DATE_RANGE_PICKER.TO";
 
 	/**
-	 * Label to be displayed in the start datepicker
+	 * Minimum date of the end date picker
 	 */
 	@Input()
-	public startDateLabel = "STARK.DATE_RANGE_PICKER.FROM";
+	public endMinDate?: Date;
 
 	/**
-	 * Maximum date of the start date picker
+	 * Maximum date of the end date picker
 	 */
 	@Input()
-	public startMaxDate?: Date;
+	public endMaxDate?: Date;
+
+	/*--- SHARED CONFIGURATIONS ---*/
 
 	/**
-	 * Minimum date of the start date picker
+	 * Input to manage both start date and end date.
 	 */
 	@Input()
-	public startMinDate?: Date;
+	public set formGroup(val: FormGroup) {
+		const { startDate, endDate } = val.controls;
+		if (!(startDate instanceof FormControl && endDate instanceof FormControl)) {
+			this.logger.warn(`[${componentName}]: formGroup requires a FormControl for startDate and endDate`);
```

Comment:
Shouldn't this be an error? Otherwise the date-range-picker won't work right?

Reply:
It will still work, but their form wont manage the stark-date-picker

Reply:
Mmm, well, throwing an error should be fine... let's make it as painful as possible :p

Reply:
😂 👍